### PR TITLE
[AEA-670 AEA-752] Libp2p node add monitoring

### PR DESCRIPTION
**Pull request by lrahmani**
## Proposed changes

Add prometheus monitoring to libp2p node
- to enable it pass monitoring uri as configuration to the connection
- when prometheus monitoring is disabled, the node will print metrics to file

Metrics are:
- ACN primitive operations: dht store and dht lookup latency
- ACN main service: Register and Route latency, number of ongoing route requests, total number of route request, total number of successful route requests
- ACN delegate service: number of concurrent clients, total number of clients
- ACN relay service: total number of clients

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go [hunk 1]
```diff
@@ -50,6 +50,10 @@ import (
 	"libp2p_node/aea"
 )
 
+const (
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/utils/utils.go:53]: Why this (particular) bound?

File: packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go [hunk 2]
```diff
@@ -762,13 +891,21 @@ func (dhtPeer *DHTPeer) handleAeaNotifStream(stream network.Stream) {
...
+		duration := timer.GetTimer(start)
+		opLatencyRegister.Observe(float64(duration.Microseconds()))
 	}
 	dhtPeer.addressAnnounced = true
 }
 
 func (dhtPeer *DHTPeer) handleAeaRegisterStream(stream network.Stream) {
 	lerror, _, linfo, _ := dhtPeer.getLoggers()
 
+	nbrClients, _ := dhtPeer.monitor.GetCounter("service_relay_clients_count_all")
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go:903]: This strings could be defined as constants elsewhere so we aid reuse and avoid magic values

File: packages/fetchai/connections/p2p_libp2p/connection.py [hunk 3]
```diff
@@ -548,6 +559,10 @@ def __init__(self, **kwargs):
         if libp2p_delegate_uri is not None:
             delegate_uri = Uri(libp2p_delegate_uri)
 
+        monitoring_uri = None
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/connection.py:562]: monitoring_uri: Optional[Uri] = None

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go (hunk 1 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/utils/utils.go:53]: it is arbitrary considering the regular messages that agents currently exchange

Comment (DavidMinarsch) [reply packages/fetchai/connections/p2p_libp2p/utils/utils.go:53]: We should keep track of this choice in docs

**Pull request merged**